internal/provider/incident_workflow_resource.go

```diff
@@ -0,0 +1,210 @@
+package provider
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+	"github.com/incident-io/terraform-provider-incident/internal/apischema"
+	"github.com/incident-io/terraform-provider-incident/internal/client"
+)
+
+var (
+	_ resource.Resource                = &IncidentWorkflowResource{}
+	_ resource.ResourceWithImportState = &IncidentWorkflowResource{}
+)
+
+type IncidentWorkflowResource struct {
+	client *client.ClientWithResponses
+}
+
+func NewIncidentWorkflowResource() resource.Resource {
+	return &IncidentWorkflowResource{}
+}
+
+type IncidentWorkflowResourceModel struct {
+	ID               types.String `tfsdk:"id"`
+	Name             types.String `tfsdk:"name"`
+	Folder           types.String `tfsdk:"folder"`
+	Trigger          types.String `tfsdk:"trigger"`
+	TerraformRepoURL types.String `tfsdk:"terraform_repo_url"`
+}
+
+func (r *IncidentWorkflowResource) Metadata(ctx context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_workflow"
+}
+
+func (r *IncidentWorkflowResource) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		MarkdownDescription: apischema.TagDocstring("Workflows V2"),
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
+				Computed: true,
+			},
+			"name": schema.StringAttribute{
+				Required: true,
+			},
+			"folder": schema.StringAttribute{
+				Optional: true,
+			},
+			"trigger": schema.StringAttribute{
```

Comment:
This can't be changed once set (well, it requires a delete), we can encode that using a plan modifier:
```
PlanModifiers: []planmodifier.String{
				stringplanmodifier.RequiresReplace(),
			},
```